src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/AppealReferenceNumberGenerator.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.service;
+
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.AsylumAppealType.from;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacaseapi.domain.DateProvider;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.AppealReferenceNumber;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.AsylumAppealType;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.AsylumCaseRetrievalException;
```

Comment:
Can we avoid calling `infrastructure` code from the domain?
The domain shouldn't know the infrastructure exists.

Reply:
Ok, I had thought you said that the pattern was that the domain and infra should communicate via interfaces, which is why I introduced the interface for the initializer, again lets discuss hen you are in

Reply:
Exception moved to domain